app/Policies/MemoryNotePolicy.php

```diff
@@ -25,36 +28,55 @@ public function viewAny(User $user): bool
         return true;
     }
 
-    public function view(User $user, GameHash $gameHash): bool
+    public function view(User $user, MemoryNote $memoryNote): bool
     {
         return true;
     }
 
     public function create(User $user): bool
     {
+        // TODO add rules for DEVELOPER_JUNIOR
+
         return $user->hasAnyRole([
+            Role::DEVELOPER_STAFF,
             Role::DEVELOPER,
         ]);
     }
 
-    public function update(User $user, GameHash $gameHash): bool
+    public function update(User $user, MemoryNote $memoryNote): bool
     {
-        return $user->hasAnyRole([
-            Role::DEVELOPER,
-        ]);
+        if ($user->hasAnyRole([Role::DEVELOPER_STAFF, Role::DEVELOPER])) {
+            return true;
+        }
+
+        // Users with the DEVELOPER_JUNIOR role are allowed to update their own notes.
+        if ($user->hasRole(Role::DEVELOPER_JUNIOR) && $user->is($memoryNote->user)) {
+            return true;
+        }
+
+        return false;
```

Comment:
Are you intentionally ignoring $user->Permissions? The Edit button is not available for existing Devs currently.

Reply:
Not intentional, this was an oversight on my part. Addressed in latest.